app/custom/page.tsx

```diff
@@ -1,20 +1,30 @@
 import { VideoConferenceClientImpl } from './VideoConferenceClientImpl';
 
-interface SearchParams {
-  liveKitUrl?: string;
-  token?: string;
-  codec?: string;
-}
-
-export default function CustomRoomConnection(props: { searchParams: SearchParams }) {
+export default function CustomRoomConnection(props: {
+  searchParams: {
+    liveKitUrl?: string;
+    token?: string;
+    codec?: string;
+  };
+}) {
   const { liveKitUrl, token, codec } = props.searchParams;
-
   if (typeof liveKitUrl !== 'string') {
     return <h2>Missing LiveKit URL</h2>;
   }
   if (typeof token !== 'string') {
     return <h2>Missing LiveKit token</h2>;
   }
+  if (
+    !(
+      codec === 'vp8' ||
```

Comment:
nit: 
`videoCodecs.includes(codec)` (we export the supported video codecs from livekit-client)

Reply:
hm, maybe you can reuse the validate helper here and make it a typeguard?